### PR TITLE
SI-6762 rename emptyValDef to emptySelfType.

### PR DESCRIPTION
**Pull request by paulp**
Looks like emptyValDef.isEmpty was already changed to return
false, so now all that's left is a name which means something.

### CODE REVIEW

**Comment by paulp** [comment]
Review by @xeno-by

**Comment by xeno-by** [comment]
How about `noSelfType`? That would avoid the potential confusion of predicting what `emptySelfType.isEmpty` should evaluate to.

**Comment by paulp** [comment]
Another fine day at the productivity factory.

``` diff
 diff --git i/test/files/run/repl-power.check w/test/files/run/repl-power.check
 index 575d94211c..c1992e89aa 100644
 --- i/test/files/run/repl-power.check
 +++ w/test/files/run/repl-power.check
 @@ -31,4 +31,4 @@ m: $r.treedsl.global.Match =
  scala> typed(m).tpe                              // typed is in scope
  res2: $r.treedsl.global.Type = Boolean

 -scala>
 +scala> 
```

It doesn't display for me, but imagine that trailing space.

File: src/reflect/scala/reflect/api/Trees.scala [hunk 1]
```diff
@@ -72,12 +72,12 @@ trait Trees { self: Universe =>
     /** Is this tree one of the empty trees?
      *
      *  Empty trees are: the `EmptyTree` null object, `TypeTree` instances that don't carry a type
-     *  and the special `emptyValDef` singleton.
+     *  and the special `noSelfType` singleton.
```
Comment (xeno-by) [inline src/reflect/scala/reflect/api/Trees.scala]: Looks like this comment can go.

File: src/reflect/scala/reflect/api/Trees.scala [hunk 2]
```diff
@@ -72,12 +72,12 @@ trait Trees { self: Universe =>
...
      *
      *  In the compiler the `isEmpty` check and the derived `orElse` method are mostly used
      *  as a check for a tree being a null object (`EmptyTree` for term trees and empty TypeTree for type trees).
      *
-     *  Unfortunately `emptyValDef` is also considered to be `isEmpty`, but this is deemed to be
+     *  Unfortunately `noSelfType` is also considered to be `isEmpty`, but this is deemed to be
      *  a conceptual mistake pending a fix in https://issues.scala-lang.org/browse/SI-6762.
```
Comment (xeno-by) [inline src/reflect/scala/reflect/api/Trees.scala]: And this one too.

File: src/reflect/scala/reflect/api/Trees.scala [hunk 3]
```diff
@@ -2121,6 +2121,9 @@ trait Trees { self: Universe =>
    *  no definition of a self value of self type.
    *  @group Trees
    */
+  val noSelfType: ValDef
+
+  @deprecated("Use noSelfType", "2.11.0")
```
Comment (xeno-by) [inline src/reflect/scala/reflect/api/Trees.scala]: Looks like most of our deprecation messages say `Use ... instead`.

File: src/reflect/scala/reflect/internal/StdNames.scala [hunk 4]
```diff
@@ -661,7 +661,7 @@ trait StdNames {
     val dollarScope: NameType          = "$scope"
     val drop: NameType                 = "drop"
     val elem: NameType                 = "elem"
-    val emptyValDef: NameType          = "emptyValDef"
+    val noSelfType: NameType        = "noSelfType"
```
Comment (xeno-by) [inline src/reflect/scala/reflect/internal/StdNames.scala]: Alignment got skewed.

**Comment by paulp** [comment]
Since if you made me tweak this one more time I would be forced to fly to where you are and throttle you, I sense that this must LGTY.

**Comment by xeno-by** [comment]
Yes, that was an lgtm.

**Pull request merged**